public/js/components/SourcesTree.js

```diff
@@ -30,33 +47,35 @@ let SourcesTree = React.createClass({
   },
 
   componentWillReceiveProps(nextProps) {
-    if (nextProps.sources !== this.props.sources) {
-      if (nextProps.sources.size === 0) {
-        this.setState(createTree(nextProps.sources));
-        return;
-      }
+    if (nextProps.sources === this.props.sources) {
```

Comment:
flipped the test so that it could be a guard statement and avoid one level of nesting.